Orange/widgets/model/owneuralnetwork.py

```diff
@@ -154,20 +163,33 @@ def setup_layout(self):
         gui.button(self.apply_button, self, "Cancel", callback=self.cancel)
 
     def create_learner(self):
-        return self.LEARNER(
-            hidden_layer_sizes=self.get_hidden_layers(),
-            activation=self.activation[self.activation_index],
-            solver=self.solver[self.solver_index],
-            alpha=self.alpha,
-            max_iter=self.max_iterations,
-            preprocessors=self.preprocessors)
+        learner = None
+        if (self.random_state > 0):
+            learner = self.LEARNER(
+                hidden_layer_sizes=self.get_hidden_layers(),
+                activation=self.activation[self.activation_index],
+                solver=self.solver[self.solver_index],
+                alpha=self.alpha,
+                max_iter=self.max_iterations,
+                preprocessors=self.preprocessors,
+                random_state=self.random_state)
+        else:
+            learner = self.LEARNER(
+                hidden_layer_sizes=self.get_hidden_layers(),
+                activation=self.activation[self.activation_index],
+                solver=self.solver[self.solver_index],
+                alpha=self.alpha,
+                max_iter=self.max_iterations,
+                preprocessors=self.preprocessors)
+        return learner
```

Comment:
These two calls are the same, except that the second uses the default values for `random_state`. You should have a single call with a variable parameter.

Reply:
Exactly. This is the reason for the conditional statement here. As I don't see a way to access the random state from the underlying library, the easiest way is to have two different calls. One with the fixed random.state and the other without it. Or you fix it in the random.state setup. For me this is more clear, than messing with hidden object states.

Reply:
You pass the value of self.random_state here (probably 0) or None if the checkbox is off.

Reply:
I see.